molgenis-data-vcf/src/main/java/org/molgenis/data/vcf/utils/VcfUtils.java

```diff
@@ -421,76 +341,67 @@ private static String toVcfDataType(MolgenisFieldTypes.FieldTypeEnum dataType)
 	 * Get pedigree data from VCF Now only support child, father, mother No fancy data structure either Output:
 	 * result.put(childID, Arrays.asList(new String[]{motherID, fatherID}));
 	 *
-	 * @param inputVcfFile
+	 * @param inputVcfFileScanner
 	 * @return
 	 * @throws FileNotFoundException
 	 */
-	public static HashMap<String, Trio> getPedigree(File inputVcfFile) throws FileNotFoundException
+	public static HashMap<String, Trio> getPedigree(Scanner inputVcfFileScanner) throws FileNotFoundException
 	{
-		HashMap<String, Trio> result = new HashMap<String, Trio>();
-
-		Scanner inputVcfFileScanner = new Scanner(inputVcfFile, "UTF-8");
-		String line = inputVcfFileScanner.nextLine();
+		HashMap<String, Trio> result = new HashMap<>();
 
-		// if first line does not start with ##, we don't trust this file as VCF
-		if (line.startsWith(VcfRepository.PREFIX))
+		while (inputVcfFileScanner.hasNextLine())
 		{
-			while (inputVcfFileScanner.hasNextLine())
+			String line = inputVcfFileScanner.nextLine();
+
+			// quit when we don't see header lines anymore
+			if (!line.startsWith(VcfRepository.PREFIX))
+			{
+				break;
+			}
+
+			// detect pedigree line
+			// expecting e.g. ##PEDIGREE=<Child=100400,Mother=100402,Father=100401>
+			if (line.startsWith("##PEDIGREE"))
 			{
-				// detect pedigree line
-				// expecting: ##PEDIGREE=<Child=100400,Mother=100402,Father=100401>
-				if (line.startsWith("##PEDIGREE"))
+				System.out.println("Pedigree data line: " + line);
```

Comment:
are the  System.out.println( for the command-line? it is slowing the web version.